data_pipeline_api/registry/access_upload.py

```diff
@@ -68,7 +69,38 @@ def _get_input_url(
     }
     object_component = get_data(query_data, DataRegistryTarget.object_component, data_registry_url, token)
     url = object_component["url"]
-    logger.info(f"Retrieved {url} for {namespace}/{data_product_name}/{component_name}/{version}")
+    logger.info(f"Retrieved {url} for {namespace}/{data_product_name}/{version}/{component_name}")
+    return url
+
+
+def _get_external_object_url(
+    external_object_identifier: str, version: str, component_name: str, data_registry_url: str, token: str
```

Comment:
I'd be inclined to call this `doi_or_unique_name` rather than `external_object_identifier` for clarity

Reply:
done